packages/loader/driver-utils/src/network.ts

```diff
@@ -80,7 +80,7 @@ export class NonRetryableError<T> extends NetworkErrorBasic<T> {
 /**
  * Throttling error class - used to communicate all throttling errors
  */
-export class ThrottlingError extends LoggingError implements IThrottlingWarning {
+export class ThrottlingError extends FluidError implements IThrottlingWarning {
```

Comment:
Note that we do add properties here and there to error objects, like in createOdspNetworkError:
```
    const errorAsAny = error as any;

    errorAsAny.response = responseText;
    if (response) {
        errorAsAny.responseType = response.type;
        if (response.headers) {
            errorAsAny.sprequestguid = nullToUndefined(response.headers.get("sprequestguid"));
            errorAsAny.serverEpoch = nullToUndefined(response.headers.get("x-fluid-epoch"));
        }
    }
```

We would need to find all these places and fix them.
Which I'd suggest to do as a pre-cursor to this change (i.e. add method addProperty() to LoggingError, change interfaces IDriverErrorBase, IOdspError, IErrorBase to expose addProperty() method and change all places where we are patching error object).

Reply:
Yes I had this exact spot in mind when I included `addDetails`.  cc @anthony-murphy as FYI.